README.md

```diff
@@ -63,43 +64,44 @@ The system consists of the following components:
    chaincodes. A chaincode library acts as intermediary between the chaincode
    in the enclave and the peer. The chaincode enclave exposes the Fabric
    chaincode interface and extends it with additional support for state
-   encryption, attestation, and secure blockchain state access. This component
-   is devided into two subcomponents: ``ecc_enclave`` contains the code
-   running inside an enclave and ``ecc`` contains a wrapper chaincode that
-   invokes the enclave.
-
-1. *Ledger enclave:* The ledger enclave maintains the ledger in an enclave in
-   the form of integrity-specific metadata representing the most recent
-   blockchain state at the peer. It performs the same validation steps as the
-   peer when a new block arrives, but additionally generates a cryptographic
-   hash of each key-value pair of the blockchain state and stores it within
-   the enclave. The ledger enclave exposes an interface to the chaincode
-   enclave for accessing the integrity-specific metadata. This is used to
-   verify the correctness of the data retrieved from the blockchain
-   state. Like the chaincode enclave, the ledger enclave is divided into two
-   subcomponents: ``tlcc`` and ``tlcc_enclave``.
-
-1. *Enclave registry:* The enclave registry is a chaincode that runs outside
+   encryption, attestation, and secure blockchain state access. This
+   code is executed inside an SGX TEE.
+
+1. *Enclave Endorsement validation:* The enclave endorsement validation
+   complements the peer’s validation system and is responsible for
+   validating transactions produced by a chaincode enclave. In
+   particular, the validator checks that a transaction contains a
+   valid signature issued by a registered chaincode enclave. If the
+   validation is successful, it marks the transactions as valid and
+   hands it over to the ledger enclave, which crosschecks the decision
+   before it finally commits the transaction to the ledger. This code
```

Comment:
```suggestion
   valid signature issued by a registered chaincode enclave. If the
   validation is successful, it marks the transactions as valid and
   transforms the chaincode state updates to a Fabric read/writeset,
   which is applied during the validation-and-commit phase. This code
```

Reply:
good that you found that as it makes me realize there are lots of other references to ledger enclave which also have to be fixed ..